WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -144,6 +145,8 @@ await roundStatusUpdater
 				}
 			}
 			while (!await TryConfirmConnectionAsync(amountsToRequest, vsizesToRequest, cancellationToken).ConfigureAwait(false));
+
+			SmartCoin.CoinJoinInCriticalPhase = true;
```

Comment:
imho there is missing lock here - so writes to `CoinJoinInCriticalPhase` are without lock (which they should if they are shared resource) and reads are with a lock (but that lock belongs to `WalletManager` which looks very fishy to me).